File: solutions/345.reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        char[] chars = s.toCharArray();
        int start = 0, end = chars.length - 1;
        while(start < end){
            if(!isVowels(chars[start])){
                start++;
                continue;
            }
            if(!isVowels(chars[end])){
                end--;
                continue;
            }
            char temp = chars[end];
            chars[end] = chars[start];
            chars[start] = temp;
            end--;
            start++;
        }
        return new String(chars);
        
    }
    public boolean isVowels(char c){
        c = (""+c).toLowerCase().charAt(0);
        if(c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u') return true;
        return false;
    }
}

File: solutions/345.reverse-vowels-of-a-string/reverse-vowels-of-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {
    const chars = s.split('');
    let start = 0, end = s.length - 1;
    while(start < end){
        if(!isVowels(chars[start])){
            start++;
        }else if(!isVowels(chars[end])){
            end--;
        }else{
            const temp = chars[start];
            chars[start] = chars[end];
            chars[end] = temp
            end--;
            start++;
        }
    }
    return chars.join('');
    
    
    function isVowels(c){
        let res = false;
        switch(c){
            case 'A':
            case 'a':
            case 'E':
            case 'e':
            case 'I':
            case 'i':
            case 'O':
            case 'o':
            case 'U':
            case 'u':
                res = true;
                break;
            default:
                break;
        }
        return res;
    }
    
};